### DISCUSSION TITLE
Performance suggestion: TimeSpan.From* complie as const

### DISCUSSION THREAD

**Discussion by Meister1977** [dotnet/runtime#94995] [Ideas] [answered]
I think, compiler can optimize  TimeSpan.From* calls to insert the value as const into the binary instead of calling the methods.

**Comment by KeterSCP** [comment, marked as answer]
JIT already optimizes this kind of pattern to simple `mov` with a constant value.
[Sharplab link](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgAYBYAKDIEYaaAVAS3xgGUAHbAOwAIA4jAwA1bABsArjAAUASn4BeAHz9W7bnwB0AMSgR8AEWwBPXLIDM8gNxA)

**Reply by tannergooding** [reply]
Noting that what the user is asking for may be something more akin to `constexpr`: https://github.com/dotnet/csharplang/discussions/2379

There is no language support for that today and such support is incredibly complex, bringing along numerous considerations and potential implications.

That all being said, the JIT (and by extension AOT) do the right thing and so your machine code ends up being already optimal today.

**Reply by Meister1977** [reply]
@KeterSCP , thanks, I did not know how I can check the binary after JIT/AOT, so I had checked only wilt ILSpy. But now, I understand it better. Thanks again your answer.
